edgeless_systemtests/src/lib.rs

```diff
@@ -0,0 +1,288 @@
+        for domain_i in 0..num_domains {
+            let orchestrator_url = format!("http://{}:{}", address, next_port());
+            let domain_id = format!("domain-{}", domain_i);
+            orchestrators.push(edgeless_con::EdgelessConOrcConfig {
+                domain_id: domain_id.clone(),
+                orchestrator_url: orchestrator_url.clone(),
+            });
+
+            let (task, handle) = futures::future::abortable(edgeless_orc::edgeless_orc_main(edgeless_orc::EdgelessOrcSettings {
+                domain_id: domain_id.to_string(),
+                orchestrator_url: orchestrator_url.to_string(),
+                orchestration_strategy: edgeless_orc::OrchestrationStrategy::Random,
+                keep_alive_interval_secs: 1,
+            }));
+            tokio::spawn(task);
+            handles.push(handle);
+
+            // The first node in each domain is also assigned a file-log resource.
+            for node_i in 0..num_nodes_per_domain {
+                let (task, handle) = futures::future::abortable(edgeless_node::edgeless_node_main(match node_i {
+                    0 => edgeless_node::EdgelessNodeSettings {
+                        node_id: uuid::Uuid::new_v4(),
+                        agent_url: format!("http://{}:{}", address, next_port()),
+                        invocation_url: format!("http://{}:{}", address, next_port()),
+                        metrics_url: format!("http://{}:{}", address, next_port()),
+                        orchestrator_url: orchestrator_url.to_string(),
+                        http_ingress_url: "".to_string(),
+                        http_ingress_provider: "".to_string(),
+                        http_egress_provider: "".to_string(),
+                        file_log_provider: "file-log-1".to_string(),
+                        redis_provider: "".to_string(),
+                    },
+                    _ => {
+                        edgeless_node::EdgelessNodeSettings::new_without_resources(&orchestrator_url, address, next_port(), next_port(), next_port())
+                    }
+                }));
+                tokio::spawn(task);
+                handles.push(handle);
+            }
+        }
+
+        let (task, handle) = futures::future::abortable(edgeless_con::edgeless_con_main(edgeless_con::EdgelessConSettings {
+            controller_url: controller_url.clone(),
+            orchestrators,
+        }));
+        tokio::spawn(task);
+        handles.push(handle);
+
+        let mut con_client = edgeless_api::grpc_impl::controller::ControllerAPIClient::new(controller_url.as_str()).await;
+
+        tokio::time::sleep(tokio::time::Duration::from_millis(100)).await;
+
+        (handles, con_client.workflow_instance_api())
+    }
+
+    async fn wf_list(client: &mut Box<(dyn WorkflowInstanceAPI)>) -> Vec<edgeless_api::workflow_instance::WorkflowInstance> {
+        match client.list(edgeless_api::workflow_instance::WorkflowId::none()).await {
```

Comment:
Not part of this PR, but we should not use `none` as a list all value and replace it with something more semantic (e.g. a separate method or an option)

Reply:
Uhm... I don't like the separate method alternative, because adding methods increases the maintenance costs much more than changing a field in a message.

But we can surely have a richer message, e.g., to support querying workflows that match some characteristics.

My general approach to things is: start with the simplest approach, then make it more complex _as needed_.